src/dune_rules/test_rules.ml

```diff
@@ -1,5 +1,32 @@
 open Import
 
+let configurations ~sctx ~dir modes =
+  let open Memo.O in
+  let+ l =
+    Memo.sequential_map
+      (Dune_file.Executables.Link_mode.Map.to_list modes)
+      ~f:(fun ((mode : Dune_file.Executables.Link_mode.t), _) ->
+        match mode with
+        | Byte_complete | Other { kind = Exe; mode = Native | Best } ->
+          Memo.return [ `exe, ".exe" ]
+        | Other { kind = Exe; mode = Byte } -> Memo.return [ `bc, ".bc" ]
+        | Other { kind = C | Object | Shared_object | Plugin; _ } ->
+          (* We don't know how to run tests in these cases *)
+          Memo.return []
+        | Other { kind = Js; _ } ->
+          let+ js_of_ocaml = Super_context.env_node sctx ~dir >>= Env_node.js_of_ocaml in
+          let multiple_targets =
+            match js_of_ocaml.targets with
+            | Some targets -> List.length (Js_of_ocaml.Target.Set.to_list targets) > 1
+            | None -> false
+          in
+          if multiple_targets
```

Comment:
I don't really understand the logic here, why do we have Js_of_ocaml.Ext.exe when targets is wasm only ?
This kind of pattern seems fragile/ error prone if with ever add a new target

Reply:
I would like to be able to simply switch between producing a JavaScript file or a Wasm binary by using a dune profile. Since we are using the JavaScript file names in dune files or in HTML files, it seems natural to use the same JavaScript file name in both cases.

Since Wasm is not widely available, one also needs to be able to generate both a Javascript file and a Wasm binary, so that a web server can send one or the other based on the browser. In this case, we need two different file names.

Do you see a better way to deal with both of these use cases?